pkg/trie/trie.go

```diff
@@ -54,13 +63,21 @@ type Hashable interface {
 	Hash() (common.Hash, error)
 }
 
-type Trie interface {
-	PrefixTrie
-	KVStore
+type ReadOnlyTrie interface {
+	fmt.Stringer
+
+	KVSRead
 	Hashable
-	ChildTrieSupport
+	ChildTriesRead
+	PrefixTrieRead
 	TrieIterator
 	TrieDeltas
```

Comment:
I think we don't need to track deltas in a `ReadOnlyTrie`

Reply:
True!